yt_dlp/extractor/picarto.py

```diff
@@ -43,16 +43,24 @@ def _real_extract(self, url):
     url
   }
 }''' % (channel_id, channel_id),  # noqa: UP031
-            })['data']
+            },
+            headers={
+                'Accept': '*/*',
+                'Content-Type': 'application/json',
+            },
+        )['data']
         metadata = data['channel']
 
         if metadata.get('online') == 0:
             raise ExtractorError('Stream is offline', expected=True)
         title = metadata['title']
 
+        url = data['getLoadBalancerUrl']['url']
+        https_url = url.replace('http', 'https', 1) if url.startswith('http:') else url
         cdn_data = self._download_json(
-            data['getLoadBalancerUrl']['url'] + '/stream/json_' + metadata['stream_name'] + '.js',
-            channel_id, 'Downloading load balancing info')
+            https_url + '/stream/json_' + metadata['stream_name'] + '.js',
+            channel_id, 'Downloading load balancing info',
+        )
```

Comment:
I would import `update_url` from utils and do this:
```py
        cdn_data = self._download_json(''.join((
            update_url(data['getLoadBalancerUrl']['url'], scheme='https'),
            '/stream/json_', metadata['stream_name'], '.js')),
            channel_id, 'Downloading load balancing info')
```

Reply:
Thanks! Fixed this.